packages/flutter/lib/src/rendering/sliver_custom_extent_list.dart

```diff
@@ -0,0 +1,302 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:developer' show Timeline;
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+
+import 'box.dart';
+import 'debug.dart';
+import 'sliver.dart';
+import 'sliver_multi_box_adaptor.dart';
+
+/// Assists [RenderSliverCustomExtentList] to preform layout, offering the
+/// correlative calculation about scroll offset and child index.
+abstract class SliverCustomExtentListAssistant {
```

Comment:
Is there no basic assistant that is ready for developers to use? Or will developers need to always create their own?